cloud/google/serviceaccount.go

```diff
@@ -26,68 +23,142 @@ import (
 )
 
 const (
-	ServiceAccountPrefix    = "k8s-machine-controller-"
-	ServiceAccount          = "service-account"
-	MachineControllerSecret = "machine-controller-credential"
+	MasterNodeServiceAccountPrefix        = "k8s-master"
+	WorkerNodeServiceAccountPrefix        = "k8s-worker"
+	MachineControllerServiceAccountPrefix = "k8s-machine-controller"
+	MachineControllerSecret               = "machine-controller-credential"
+	ClusterAnnotationPrefix               = "service-account-"
```

Comment:
We might want to adopt a provider specific prefix. Like `gce.clusterapi.k8s.io/service-account-`.  WDYT?

Reply:
good idea. Added